### ISSUE TITLE
SA1217 asks for incorrect sorting with System usings

### ISSUE THREAD

**Issue by RobinHood70** [DotNetAnalyzers/StyleCopAnalyzers#2163]
This was discussed in pull #786, but it doesn't seem to have been addressed.

If you ask VS to sort usings (Edit, Intellisense, Organize Usings), assuming you have the option set in your preferences, it will sort them with "using static System.*" on top of other "using static"s. StyleCop, however, is asking for that to be alphabetical.

**Comment by dlemstra** [comment]
@RobinHood70 Do you have an example so we could add unit test for this and try to fix it?

**Comment by RobinHood70** [comment]
In one of my projects, I have the following mess of usings:

	using System;
	using System.Globalization;
	using System.IO;
	using System.IO.Compression;
	using System.Net;
	using System.Net.Configuration;
	using System.Reflection;
	using System.Runtime.Serialization.Formatters.Binary;
	using System.Text;
	using System.Threading;
	using Design;
	using WikiCommon;
	using static System.Net.HttpStatusCode;
	using static ClientShared;
	using static Properties.Messages;
	using static WikiCommon.Globals;

If you do a Remove and Sort Usings, this is the order they appear in. However, SA1217 triggers in the System.Net.HttpStatusCode because it's not in alphabetical order. This behaviour differs from the regular usings rule, which allows System usings to be sorted first (as they are above).

**Comment by commonsensesoftware** [comment]
Looks like this is still up for someone to update. I've also noticed that this rule (SA1217) conflicts with SA1208 in some cases. Visual Studio has a setting to do System first or alphabetical. This rule can also be realized in .editorconfig with the `dotnet_sort_system_directives_first = true` setting. However, there is no setting to sort `using` one way and `using static` another way. This means that the two rules conflict unless all `using` statements are sorted alphabetically.

Personally, I find myself using the _Remove and Sort Usings_ (CTRL+R, CTRL+G) shortcut a lot now, but I always end up having to manually fix the `using static` because the SA1217 says they can only be alphabetic. While I prefer the simplicity of alphabetic only, I work with teams where that's not always the established practice. I think the rules should be congruent; one way or the other.

### System First
 ```editorconfig
dotnet_sort_system_directives_first = true
```
```c#
using System;
using System.IO;
using Microsoft.Security;
using static System.Math;
using static Microsoft.Security.IdentityType;
```

### All Alphabetic
 ```editorconfig
dotnet_sort_system_directives_first = false
```
```c#
using Microsoft.Security;
using System;
using System.IO;
using static Microsoft.Security.IdentityType;
using static System.Math;
```

**Comment by lostindark** [comment]
Ran into the same issue as well. Given VS 2017 can automatically do code clean up upon save (sort using), this will cause more issues. The fix should be prioritized.
